processing/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -4041,7 +4042,9 @@ public Function asSingleThreaded(List<Expr> args, Expr.InputBindingInspector ins
           final Object[] rhsArray = rhsEval.castTo(lhsArrayType).asArray();
           return new ContainsConstantArray(rhsArray);
         } else {
-          final Object val = rhsEval.castTo((ExpressionType) lhsArrayType.getElementType()).value();
+          final ExpressionType lhsElementType = (ExpressionType) ObjectUtils.defaultIfNull(
```

Comment:
I think this part is not needed anymore, according to a response from a druid member, but we can leave it as is IMO